From e44d7ef87af8af1781ffba5e3c5606186eac1d4b Mon Sep 17 00:00:00 2001
From: "brian.wagner" <brian.wagner@lairdtech.com>
Date: Mon, 15 Feb 2016 14:16:01 -0500
Subject: [PATCH] laird: ath6kl: Resolve kernel warning on ath6kl_core remove

added cfg80211_disconnected to properly clear bss before unregister_netdevice
bug 8784
---
 drivers/net/wireless/ath/ath6kl/cfg80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath6kl/cfg80211.c b/drivers/net/wireless/ath/ath6kl/cfg80211.c
index 3220436dec31..6a19cdae201c 100644
--- a/drivers/net/wireless/ath/ath6kl/cfg80211.c
+++ b/drivers/net/wireless/ath/ath6kl/cfg80211.c
@@ -3664,6 +3664,7 @@ void ath6kl_cfg80211_vif_cleanup(struct ath6kl_vif *vif)
 		kfree(mc_filter);
 	}
 
+	cfg80211_disconnected(vif->ndev, 0, NULL, 0, true, GFP_KERNEL);
 	unregister_netdevice(vif->ndev);
 
 	ar->num_vif--;
-- 
2.11.0

